clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -656,18 +670,20 @@ BuiltinTypeDeclBuilder::addSimpleTemplateParams(ArrayRef<StringRef> Names,
 }
 
 BuiltinTypeDeclBuilder &BuiltinTypeDeclBuilder::addIncrementCounterMethod() {
+  using PH = BuiltinTypeMethodBuilder::PlaceHolder;
   return BuiltinTypeMethodBuilder(SemaRef, *this, "IncrementCounter",
                                   SemaRef.getASTContext().UnsignedIntTy)
-      .callBuiltin("__builtin_hlsl_buffer_update_counter",
-                   {getConstantIntExpr(1)})
+      .callBuiltin("__builtin_hlsl_buffer_update_counter", PH::Handle,
+                   getConstantIntExpr(1))
```

Comment:
How evil would it be to add a `Expr *convertPlaceholder(int)` that does the `getConstantIntExpr` for you, so this could be written as: `.callBuiltin("__builtin_hlsl_buffer_update_conter", PH::Handle, 1)`?

Reply:
This feels too magical to me (and also it introduces an annoying ambiguity with a literal zero - is it an `int` or a null `Expr *`?)